praw/models/comment_forest.py

```diff
@@ -72,6 +72,12 @@ def __len__(self) -> int:
         """Return the number of top-level comments in the forest."""
         return len(self._comments)
 
+    def __eq__(self, other: _CommentForest):
+        """Compare two comment forest instances."""
+        if isinstance(other, CommentForest):
+            return self.list() == other.list()
```

Comment:
Is this the best way to do it? If I'm not mistaken, calling `.list()` will trigger a lazy load if the comments haven't been fetched yet. That seems unnecessary for just checking equality.

Reply:
How can the replies be compared if there are no replies?